mentoring/mentoring.py

```diff
@@ -25,70 +25,226 @@
+        help="Has the student completed this mentoring step?",
+        default=False,
+        scope=Scope.user_state
+    )
+    num_attempts = Integer(
+        help="Number of attempts a user has answered for this questions",
+        default=0,
+        scope=Scope.user_state,
+        enforce_type=True
+    )
+    step = Integer(
+        help="Keep track of the student assessment progress.",
+        default=0,
+        scope=Scope.user_state,
+        enforce_type=True
+    )
+    student_results = List(
+        help="Store results of student choices.",
+        default=[],
+        scope=Scope.user_state
+    )
+
+    # Global user state
+    next_step = String(
+        help="url_name of the next step the student must complete (global to all blocks)",
+        default='mentoring_first',
+        scope=Scope.preferences
+    )
+
+    editable_fields = (
+        'mode', 'followed_by', 'max_attempts', 'enforce_dependency',
+        'display_submit', 'weight', 'display_name',
+    )
     icon_class = 'problem'
+    has_score = True
+    has_children = True
+
+    def _(self, text):
+        """ translate text """
+        return self.runtime.service(self, "i18n").ugettext(text)
+
+    @property
+    def is_assessment(self):
+        return self.mode == 'assessment'
+
+    @property
+    def score(self):
+        """Compute the student score taking into account the weight of each step."""
+        weights = (float(self.runtime.get_block(step_id).weight) for step_id in self.steps)
+        total_child_weight = sum(weights)
+        if total_child_weight == 0:
+            return Score(0, 0, 0, 0, 0)
+        score = sum(r[1]['score'] * r[1]['weight'] for r in self.student_results) / total_child_weight
+        correct = sum(1 for r in self.student_results if r[1]['status'] == 'correct')
+        incorrect = sum(1 for r in self.student_results if r[1]['status'] == 'incorrect')
+        partially_correct = sum(1 for r in self.student_results if r[1]['status'] == 'partial')
+
+        return Score(score, int(round(score * 100)), correct, incorrect, partially_correct)
 
     def student_view(self, context):
-        fragment, named_children = self.get_children_fragment(context, view_name='mentoring_view',
-                                                              not_instance_of=MentoringMessageBlock)
+        # Migrate stored data if necessary
+        self.migrate_fields()
+
+        fragment = Fragment()
+        child_content = u""
 
-        correct_icon_url = self.runtime.local_resource_url(self, 'public/img/correct-icon.png')
+        for child_id in self.children:
+            child = self.runtime.get_block(child_id)
+            if isinstance(child, MentoringMessageBlock):
+                pass  # TODO
+            else:
+                try:
+                    child_fragment = child.render('mentoring_view', context)
+                except NoSuchViewError:
+                    if child.scope_ids.block_type == 'html':
```

Comment:
Why only allow one other type of XBlock? Imho we should just default to the student_view if mentoring_view can't be found, this would allow to include any type of XBlock there - like videos for example. There wouldn't be an easy way to add these from Studio, but that doesn't mean it should be forbidden to do it.